docs/dev/testing.md

```diff
@@ -0,0 +1,76 @@
+Unit tests are located next to a module named `module_name_test.go`
+
+* **Subsystem tests** - Focused on the component while mocking other component. (e.g. agent/installer results).
+
+* **System tests** (a.k.a e2e) - Running full flows with all components.
+The e2e tests are divided into u/s (upstream) basic workflows on [assisted-test-infra](https://github.com/openshift/assisted-test-infra/tree/master/discovery-infra/tests) and d/s (downstream) extended regression tests maintained by both DEV and QE teams on [kni-assisted-installer-auto](https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/tree/master/api_tests).
+
+## Repository CI
+
+| [Jenkins]((http://assisted-jenkins.usersys.redhat.com)) | [Prow](https://prow.ci.openshift.org) |
+|---|---|
+| Local for Assisted ecosystem | Company-wide |
+| Checks comments for JIRA | Runs e2e
+| Manages images in quay.io/ocpmetal | Runs all testing checks (lint, unit, etc)
+
+Assisted-service CI jobs are defined under [openshift/release](https://github.com/openshift/release) repository on [openshift-assisted-service-master.yaml](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml).
+
+All the available jobs can be viewed on [Openshift CI Step Registry](https://steps.ci.openshift.org/search?job=openshift-assisted-service).
+
+Read more about OpenShift CI infrastructure on [OpenShift CI Docs](https://docs.ci.openshift.org/docs/).
+
+### Adding a new CI job
+
+When adding a new job the following rules of thumbs should be taken into account:
+
+1. Test logic needs to be maintained in the repository under test and not under openshift/release.
+It would allow easier integration with other tools, less dependency of the CI infrastructure, and most importantly the availability to run it locally.
+
+1. When introducing a new job it should be both a presubmit and a periodic. A presubmit needs to be available so contributors would be able to run it whenever they want.
+The presubmit needs to be configured as `always_run: false` and `optional: true` (not blocking a merge) until proving stability.
+The periodic needs to run on a frequent basis (e.g. daily) and have a `reporter_config` configured, in order to be notified on Slack whenever there's a breakage.
+
+1. In case the new job affects multiple repositories - every repository should have the same presubmit job so it could be tested for every component change. For example [e2e-metal-assisted-olm](https://steps.ci.openshift.org/search?job=e2e-metal-assisted-olm).
+
+[An example of a PR adding a new job](https://github.com/openshift/release/pull/21604)
+
+### FAQ
+
+#### **How can I debug CI failures?**
+
+A CI job can be debugged only in runtime.
+Once a job terminates it can no longer be debugged.
+Follow [Debug Prow Jobs live](https://gist.github.com/omertuc/1ef4bdf22f0fedfbde46cf1feb149bb9) gist.
```

Comment:
"You can try to run the <link gist in order to connect to the OCP cluster running your prow tests"

Reply:
The gist currently tries to connect to both the OCP cluster running the prow tests AND the OCP cluster that is created by the ipi-install-install step (not sure this step is even used by any of the AI jobs). Maybe the gist should be modified to be more robust and check if the job even has ipi-install-install.

I think for now mentioned that the gist is experimental and any contributions to improve it are welcome